### PR TITLE
feat(scanner): support offset markers and nth-occurrence matching

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary by CodeRabbit

* **New Features**
  * Added an AOB offset marker (using `|`) to target a specific byte within a matched pattern.
  * Added Nth-occurrence matching to locate the 1-based occurrence of a pattern; returns no match when the requested occurrence isn’t found.

* **Documentation**
  * README updated to explain the offset marker and occurrence syntax, including behavior with wildcards and edge positions.

* **Tests**
  * New tests cover offset parsing, invalid/multiple markers, edge cases, overlapping matches, and Nth-occurrence behavior.

### CODE REVIEW

File: include/DetourModKit/scanner.hpp [hunk 1]
```diff
@@ -23,6 +23,8 @@ namespace DetourModKit
         {
             std::vector<std::byte> bytes; ///< Pattern bytes (wildcard positions contain arbitrary values)
             std::vector<std::byte> mask;  ///< 0xFF = match this byte, 0x00 = wildcard (skip)
+            size_t offset = 0;            ///< Byte offset from pattern start to the point of interest.
+                                          ///< Set by `|` marker in the AOB string, or 0 if absent.
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply include/DetourModKit/scanner.hpp:27]: The nth-occurrence overload is a thin delegation wrapper. All input validation
is already handled by the single-match find_pattern it calls on each iteration:

- start_address == nullptr: find_pattern logs error and returns nullptr, loop
  breaks immediately
- region_size == 0: while (remaining >= pattern.size()) fails on first check,
  returns nullptr
- pattern.size() == 0: find_pattern logs error and returns nullptr, loop breaks

Adding duplicate guards at the top would validate against our own internal
function, not against an external boundary. The instructions say "only validate
at system boundaries (user input, external APIs)" and "don't add error handling,
fallbacks, or validation for scenarios that can't happen." The delegation already
produces correct results and correct log messages for all edge cases.

The occurrence == 0 check IS there because that's a new parameter unique to this
overload and not covered by delegation.
